portfolio/src/main/webapp/script.js

```diff
@@ -27,3 +27,32 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+// Written with help from https://www.w3schools.com/howto/howto_html_include.asp
+function createNavigation() {
+    var total, i, element, file, xhttp;
+    // Loop through a collection of all HTML elements. 
+    total = document.getElementsByTagName("*");
+    for (i = 0; i < total.length; i++) {
+        element = total[i];
+    // Search for elements with a certain atrribute.
+    file = element.getAttribute("w3-include-html");
+    if (file) {
+        // Make an HTTP request using the attribute value as the file name
+        xhttp = new XMLHttpRequest();
+        xhttp.onreadystatechange = function() {
+        if (this.readyState == 4) {
+            if (this.status == 200) {element.innerHTML = this.responseText;}
+            if (this.status == 404) {element.innerHTML = "Page not found.";}
+            // Remove the attribute, and call this function once more
+            element.removeAttribute("w3-include-html");
+            createNavigation();
+        }
+      }
+      xhttp.open("GET", file, true);
```

Comment:
Add parameter name comments for constant parameters: https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments